UN-OCHA/common_design
# [CD-489] Dropdown improvements

## Types of changes

- :heavy_check_mark: Improvement (non-breaking change which iterates on an existing feature)
-  :heavy_check_mark: New feature (non-breaking change which adds functionality)

## Description

This refactors the way to find the toggler button by using the `aria-controls` attribute which contains the ID of the toggable element instead of assuming it is always the previousElementSibling.

This enables the possibility to add the toggler element after the toggable element for "show more" type  buttons view a new `data-cd-insert-after` attribute on the toggable element.

This has been used on production on RW for a while now for the "Show full description" button of the country or disaster descriptions, for example: https://reliefweb.int/disaster/eq-2021-000116-hti.

This is also used on UNO D10 on the `/funding` page for the "view all contributions" button.

## Steps to test

  1. Any site using the CD, update the cd-dropdown script with this PR's version.
  2. Check that the user menu or any other dropdown menu (ex: main navigation on small screen) still works.
  3. Edit a template, add the following code and check that the show/hide button works

```
    <div
      data-cd-toggable="Show"
      data-cd-toggable-expanded="Hide"
      data-cd-insert-after=""
      data-cd-replace="whatever"
    >Some text.</div>
```

## Impact

The changes are compatible with the previous versions of the dropdown and should have no impact.

If approved and merged, this would allow to remove the overrides on RW and UNO.

## PR Checklist

- [x] I have followed the Conventional Commits guidelines.
- [x] My change requires a change to the documentation. <-- README of cd-dropdown component.
- [x] I have made changes to the sub theme to reflect those in the base theme
